catalog/core/catalog-core-downloadaction/src/test/java/org/codice/ddf/catalog/download/action/ResourceDownloadActionEndpointTest.java

```diff
@@ -0,0 +1,109 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.download.action;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+import javax.management.MBeanException;
+import javax.ws.rs.core.Response;
+
+import org.codice.ddf.catalog.resource.download.DownloadToCacheException;
+import org.codice.ddf.catalog.resource.download.ResourceDownloadMBean;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ResourceDownloadActionEndpointTest {
+
+    private static final String METACARD_ID = "57a4b894e13a455b8cccb87cec778b58";
+
+    private static final String SOURCE_ID = "ddf.distribution";
+
+    private static final String SUCCESS_MESSAGE_TEMPLATE = "Successfully started download of resource associated with metacard id %s from source %s";
+
+    @Mock
+    private ResourceDownloadMBean mockResourceDownloadMBeanProxy;
+
+    @Test(expected = DownloadToCacheException.class)
+    public void testDownloadToCacheMBeanProxyThrowsException() throws Exception {
+        // Setup
+        setupMockResourceDownloadMBeanWithException();
+        ResourceDownloadActionEndpoint resourceDownloadEndpoint = createResourceDownloadActionEndpoint();
+
+        // Perform Test
+        resourceDownloadEndpoint.downloadToCache(SOURCE_ID, METACARD_ID);
+    }
+
+    @Test(expected = DownloadToCacheException.class)
+    public void testDownloadToCacheNullSourceId() throws Exception {
+        // Setup
+        setupMockResourceDownloadMBeanWithException();
+        ResourceDownloadActionEndpoint resourceDownloadEndpoint = createResourceDownloadActionEndpoint();
+
+        // Perform Test
+        resourceDownloadEndpoint.downloadToCache(null, METACARD_ID);
+    }
+
+    @Test(expected = DownloadToCacheException.class)
+    public void testDownloadToCacheNullMetacardId() throws Exception {
+        // Setup
+        setupMockResourceDownloadMBeanWithException();
+        ResourceDownloadActionEndpoint resourceDownloadEndpoint = createResourceDownloadActionEndpoint();
+
+        // Perform Test
+        resourceDownloadEndpoint.downloadToCache(SOURCE_ID, null);
+    }
+
+    @Test
+    public void testDownloadToCache() throws Exception {
+        // Setup
+        setupMockResourceDownloadMBean();
+        ResourceDownloadActionEndpoint resourceDownloadEndpoint = createResourceDownloadActionEndpoint();
+
+        // Perform Test
+        Response response = resourceDownloadEndpoint.downloadToCache(SOURCE_ID, METACARD_ID);
+
+        assertThat(response.getEntity(),
+                is(String.format(SUCCESS_MESSAGE_TEMPLATE, METACARD_ID, SOURCE_ID)));
+        assertThat(response.getStatus(), is(Response.Status.OK.getStatusCode()));
```

Comment:
Should we also verify that the `ResourceDownloadMBean` was called with the right parameters?